### PR TITLE
Adds support for encrypted API keys

### LINKED ISSUE #362: Provision production API keys
Update internal API key whitelist once production keys are created.

### PR DESCRIPTION
**Pull request by ecgreb**
Release builds can now be provisioned with encrypted API keys that will be decoded at runtime.

Closes #362

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -132,4 +132,6 @@ dependencies {
     exclude group: 'org.apache.httpcomponents', module: 'httpclient'
   }
   testCompile "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
+
+  releaseCompile 'com.mapzen.android:leyndo:0.0.1-SNAPSHOT'
```
Comment (ecgreb) [inline app/build.gradle]: Gradle tries to resolve dependencies marked as `releaseCompile` when you run `assembleDebug`. 

```
> Could not resolve all dependencies for configuration ':app:_releaseCompile'.
   > Could not find com.mapzen.android:leyndo:0.0.1-SNAPSHOT.
```

Wow.

File: scripts/encrypter.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+require "android-all-4.4_r1-robolectric-0.jar"
```
Comment (sarahlensing) [inline scripts/encrypter.rb]: when running `jruby scripts/encrypter.rb defaultsalt testkey` i get a `LoadError`

```
Sarahs-MacBook-Pro:eraser-map sarahlensing$ git status -s
?? android-all-4.4_r1-robolectric-0.jar
?? app/src/main/java/com/mapzen/erasermap/SimpleCrypt.class
Sarahs-MacBook-Pro:eraser-map sarahlensing$ jruby scripts/encrypter.rb defaultsalt testkey
LoadError: no such file to load -- android-all-4.4_r1-robolectric-0
  require at org/jruby/RubyKernel.java:937
  require at /usr/local/Cellar/jruby/9.0.5.0/libexec/lib/ruby/stdlib/rubygems/core_ext/kernel_require.rb:54
    <top> at scripts/encrypter.rb:1
```

Comment (ecgreb) [inline scripts/encrypter.rb]: Can you verify your version of jruby is set to 1.7.12. Also this script should not be invoked directly rather through `encrypter.sh` which downloads the robolectric jar.

**Comment by ecgreb** [comment]
@sarahlensing @msmollin I am going to merge this to resolve build errors currently in master due to an `on-the-road` API update. Please still add comments and any fixes will be in a follow up PR.

```
> Building 84%6% > :app:compileDebugUnitTestKotlin:app:compileDebugUnitTestKotlin
> Building 86% > :app:compileDebugUnitTestKotline: /home/ubuntu/eraser-map/app/src/test/kotlin/com/mapzen/erasermap/model/TestRouter.kt: (8, 8): Class 'TestRouter' must be declared abstract or implement abstract member public abstract fun setLogLevel(logLevel: retrofit.RestAdapter.LogLevel): com.mapzen.valhalla.Router defined in com.mapzen.valhalla.Router
```

File: scripts/encrypter.rb (hunk 2 above)
Comment (sarahlensing) [inline scripts/encrypter.rb]: I had 9.0.5.0, not 1.7.12

Looks like its working now, thanks

**Pull request merged**